### ISSUE TITLE
cudnnFindConvolutionForwardAlgorithm failed: 	2   some methods tried 

### ISSUE THREAD

**Issue by davidsonic** [soumith/dcgan.torch#67]
The error is as follows:

duanjiali@cbsr-X9DRG-QF:/data/duanjiali/program/gan$ DATA_ROOT=/data/duanjiali/data/lsun_demo dataset=lsun th main.lua 
{
  ntrain : inf
  beta1 : 0.5
  name : "experiment1"
  niter : 25
  batchSize : 32
  ndf : 64
  fineSize : 64
  nz : 100
  loadSize : 96
  gpu : 1
  ngf : 64
  dataset : "lsun"
  lr : 0.0002
  noise : "normal"
  nThreads : 4
  display_id : 10
  display : 1
}
Random Seed: 5591	
DataLoader initialized!!	
Starting donkey with id: 1 seed: 5592
table: 0x41b22358
Starting donkey with id: 2 seed: 5593
table: 0x418dc238
Starting donkey with id: 4 seed: 5595
table: 0x41096b10
Starting donkey with id: 3 seed: 5594
table: 0x41bce338
Classes:
1	bedroom
Classes:
1	bedroom
Classes:
1	bedroom
initializing train loader
initializing: 	bedroom
initializing train loader
initializing: 	bedroom
initializing train loader
initializing: 	bedroom
Classes:
1	bedroom
initializing train loader
initializing: 	bedroom
Dataset: lsun	 Size: 	3033042	

cudnnFindConvolutionForwardAlgorithm failed: 	2	 convDesc=[mode : CUDNN_CROSS_CORRELATION datatype : CUDNN_DATA_FLOAT] hash=-dimA32,64,32,32 -filtA64,3,4,4 32,3,64,64 -padA1,1 -convStrideA2,2 CUDNN_DATA_FLOAT	
/data/duanjiali/basic/torch/install/bin/luajit: ...jiali/basic/torch/install/share/lua/5.1/nn/Container.lua:67: 
In 13 module of nn.Sequential:
...anjiali/basic/torch/install/share/lua/5.1/cudnn/find.lua:483: cudnnFindConvolutionForwardAlgorithm failed, sizes:  convDesc=[mode : CUDNN_CROSS_CORRELATION datatype : CUDNN_DATA_FLOAT] hash=-dimA32,64,32,32 -filtA64,3,4,4 32,3,64,64 -padA1,1 -convStrideA2,2 CUDNN_DATA_FLOAT
stack traceback:
	[C]: in function 'error'
	...anjiali/basic/torch/install/share/lua/5.1/cudnn/find.lua:483: in function 'forwardAlgorithm'
	...h/install/share/lua/5.1/cudnn/SpatialFullConvolution.lua:123: in function 'updateGradInput'
	...uanjiali/basic/torch/install/share/lua/5.1/nn/Module.lua:31: in function <...uanjiali/basic/torch/install/share/lua/5.1/nn/Module.lua:29>
	[C]: in function 'xpcall'
	...jiali/basic/torch/install/share/lua/5.1/nn/Container.lua:63: in function 'rethrowErrors'
	...iali/basic/torch/install/share/lua/5.1/nn/Sequential.lua:84: in function 'backward'
	main.lua:203: in function 'opfunc'
	...anjiali/basic/torch/install/share/lua/5.1/optim/adam.lua:37: in function 'adam'
	main.lua:217: in main chunk
	[C]: in function 'dofile'
	...asic/torch/install/lib/luarocks/rocks/trepl/scm-1/bin/th:150: in main chunk
	[C]: at 0x00406620

WARNING: If you see a stack trace below, it doesn't point to the place where this error occurred. Please use only the one above.
stack traceback:
	[C]: in function 'error'
	...jiali/basic/torch/install/share/lua/5.1/nn/Container.lua:67: in function 'rethrowErrors'
	...iali/basic/torch/install/share/lua/5.1/nn/Sequential.lua:84: in function 'backward'
	main.lua:203: in function 'opfunc'
	...anjiali/basic/torch/install/share/lua/5.1/optim/adam.lua:37: in function 'adam'
	main.lua:217: in main chunk
	[C]: in function 'dofile'
	...asic/torch/install/lib/luarocks/rocks/trepl/scm-1/bin/th:150: in main chunk
	[C]: at 0x00406620

I've tried methods mentioned in other posts like (1) luarocks install cutorch, cudnn (2) memory not enough, so I reduced the batchSize and increase the number of iterations but none of the above methods worked. How could I solve this problem? Anyone?

**Comment by davidsonic** [comment]
Problem solved, I commented the line cutorch.setDevice() when debugging and the default 0 gpu card was occupied. So it seems that the error above was due to memory insufficiency. I did the following: (1) luarocks install cutorch (2) enough memory

**Comment by ChangshiFan** [comment]
I also had this problem. The first step have finished, but i don't know how to get enough memory?
Can you tell me ?
